flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerServicesConfiguration.java

```diff
@@ -259,6 +271,30 @@ public static TaskManagerServicesConfiguration fromConfiguration(
 
 		final RetryingRegistrationConfiguration retryingRegistrationConfiguration = RetryingRegistrationConfiguration.fromConfiguration(configuration);
 
+		if (configuration.getBoolean(TaskManagerOptions.ENABLE_FLIP_49_CONFIG)) {
+			final TaskExecutorResourceSpec taskExecutorResourceSpec = TaskExecutorResourceUtils.resourceSpecFromConfig(configuration);
```

Comment:
I'm not sure I understand this comment completely.

First of all, as mentioned above, the old parameters cannot be mapped to `TaskExecutorResourceSpec` directly.

I'd like to keep the two configuration logics separately for the following reasons:
- This PR means to introduce no behavior changes to the legacy mode, which is hard to guarantee if we try to map the old fields to the new ones. Since the legacy code path is quite complex and not friendly to maintainability (exactly one of the motivations for FLIP-49), and will be removed soon, I'd rather not to spend too much effort on deduplication of the two code paths with the risk of breaking the original behavior of the legacy mode. 
- The code path of the new mode, in my opinion, is clear and self contained. I'm in favor of not messing it by inserting the legacy logics in between. That would also introduce more workload for later legacy clean-ups.

Reply:
The idea was that if we had less code path splits now then we would have less work later when we remove/deprecate the old approach, ideally soon :). The reason is that the CI would already check that the most of the code already works for both old/new mode and we will give it exposure in E2E tests. This way we automatically guarantee that the new code is better integrated into the overall code which currently relies on legacy approach. Also, if we have the feature switch only in two places (creating `TaskExecutorResourceSpec` for container parameters and in TM config) the difference between old/new modes will be also more observable and easy to remove.

True, if `taskHeapSize` is probably worse choice for `taskManagerHeapSizeMB` than e.g. `onHeapManagedMemorySize`. If we keep all heap sizes zero except one for legacy mode then it seems new code should work for legacy already now (modulo do not add zero sizes to JVM args).

What are any other big obstacles for this approach?

Reply:
I understand the motivation of having as less as possible separated code paths, for exposing the new codes for testability early.

My key point is that, this cannot be easily done.
- The old fields cannot be mapped to the new fields. `onHeapManagedMemorySize` does not match `taskManagerHeapSizeMB` neither. The legacy filed `taskManagerHeapSizeMB` has a "heap" in its name, but is actually not describing java heap memory. I don't think any of the filed in `TaskExecutorResourceSpec` matches it.
- FLIP-49 is designed to change the way of memory configuration. We can foresee that not all of the IT/E2E cases can work as it is. It doesn't make sense to try to expose the new configuration logics to the ETE tests at this moment.
- My plan was to first get the new feature completed, then switching IT/ETE test cases to the new configuration (because we no longer need them to guard the old logic). That's why I put the work item "fix / update / remove test cases for legacy mode" in the last step "clean-up of legacy mode" of FLIP-49.